bundles/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/handler/SpotifyBridgeHandler.java

```diff
@@ -340,13 +352,22 @@ private boolean pollStatus() {
         synchronized (pollSynchronization) {
             try {
                 onAccessTokenResponse(getAccessTokenResponse());
+                // Collect currently playing context.
+                final CurrentlyPlayingContext pc = playingContextCache.getValue();
+
+                if (pc != null && pc.getDevice() == null) {
+                    // Spotify returned a 204. Meaning everything is ok, but we got no data.
+                    // Happens when no song is playing.
+                    // No need to continue because no new information will be available.
+                    updateStatus(ThingStatus.ONLINE);
```

Comment:
But should we update the device-status / player info to update that its not playing anything etc?

Reply:
doh! Now I have to think about complex exceptional cases :smile: Will have a look.